syncer/syncer.go

```diff
@@ -0,0 +1,272 @@
+package syncer
```

Comment:
Please add in-code docs for methods in this file 🙏

Reply:
Added b51dc66694731673b5516b5e78c4cf8037e56cb1